optuna/storages/cached_storage.py

```diff
@@ -98,14 +112,30 @@ def get_study_id_from_name(self, study_name: str) -> int:
 
     def get_study_id_from_trial_id(self, trial_id: int) -> int:
 
+        with self._lock:
+            if trial_id in self._trial_id_to_study_id_and_number:
+                return self._trial_id_to_study_id_and_number[trial_id][0]
+
         return self._backend.get_study_id_from_trial_id(trial_id)
 
     def get_study_name_from_id(self, study_id: int) -> str:
 
+        with self._lock:
+            if study_id in self._studies:
+                name = self._studies[study_id].name
+                if name is not None:
+                    return name
+
         return self._backend.get_study_name_from_id(study_id)
 
     def get_study_direction(self, study_id: int) -> StudyDirection:
 
+        with self._lock:
+            if study_id in self._studies:
+                direction = self._studies[study_id].direction
+                if direction != StudyDirection.NOT_SET:
```

Comment:
[MEMO] This condition can hold if a loaded study (i.e., created via `load_study()`) is used.

Reply:
It's not a strong opinion but I'm wondering that it might be good to cache the direction after retrieving it from the backend storage (line 137) to support the above `load_study()` case.
What do you think of it? > @ytsmiling

Reply:
I think we definitely cache that. Thank you for the catch and I'll add it.